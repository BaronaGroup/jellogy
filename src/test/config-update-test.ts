import {prepareTestKarhu} from './test-utils'

describe('config-update-test', () => {
  const karhuTest = prepareTestKarhu({})

  it('is possible to update configuration when running', karhuTest((karhu, output) => {
    const ctx = karhu.context('test')
    ctx.debug('This should not show up')
    karhu.reconfigure({defaultLogLevel: 'DEBUG'})
    ctx.debug('This should show up')
    expect(output.tracked).toMatchSnapshot()
  }))

  it('is not possible to update log levels', karhuTest((karhu, output) => {
    expect(() => karhu.reconfigure({logLevels: ['KEK']}))
      .toThrowErrorMatchingSnapshot()
  }))
})
