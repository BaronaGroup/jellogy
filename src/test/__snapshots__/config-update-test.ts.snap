// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config-update-test is not possible to update log levels 1`] = `"Log levels cannot be updated with reconfigure"`;

exports[`config-update-test is possible to update configuration when running 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [
    Array [
      "[1] [95mDEBUG test This should show up[39m",
      "DEBUG",
      "test",
    ],
  ],
  "INFO" => Array [],
  "WARN" => Array [],
  "ERROR" => Array [],
}
`;
